engine/src/main/java/org/terasology/world/generator/WorldGenerator.java

```diff
@@ -39,6 +42,10 @@
 
     World getWorld();
 
+    List<Zone> getZones();
+
+    Zone getNamedZone(String name);
```

Comment:
Yeah, this probably should be an optional. Currently the method's only use is where the name is known to correspond to a zone, but there are uses (specifically with using zones with world plugins) where there might not be a zone for a given name.

I have realised that the current only usage (in the `PreviewWorldScreen`) is redundant (it fetches the target zone's name, then retrieves the same zone again with this method).

Identifying with a URI is probably a good idea, especially if they need to be referenced between modules. I was just using the name as a display name for the preview, but I think there will need to be a nice way of accessing zones across modules.